src/posh-git.psd1

```diff
@@ -4,7 +4,7 @@
 RootModule = 'posh-git.psm1'
 
 # Version number of this module.
-ModuleVersion = '1.0.0.0'
+ModuleVersion = '1.0.0'
```

Comment:
It definitely needs to be `1.0.0` when it is published - see https://docs.microsoft.com/en-us/powershell/gallery/psget/module/prereleasemodule.  

I worry a bit about precedence order when a user has a prerelease installed and has cloned the repo.  But perhaps the fact that our repo dir struct doesn't lend itself to being added to `PSModulePath` works in our favor in this case.  

In trying to test the precedence order with the new SemanticVersion type in PS Core, I see that it chokes on `1.0.0.0`:
```
14:143ms> [System.Management.Automation.SemanticVersion]'1.0.0.0'
Cannot convert value "1.0.0.0" to type "System.Management.Automation.SemanticVersion". Error: "Input string was not in a correct format."
At line:1 char:1
+ [System.Management.Automation.SemanticVersion]'1.0.0.0'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [], RuntimeException
+ FullyQualifiedErrorId : InvalidCastParseTargetInvocation

[rkeithhill/refactor-write-prompt ≡] ~\GitHub\dahlbyk\posh-git
15:96ms> [System.Management.Automation.SemanticVersion]'1.0.0-beta1'

Major  Minor  Patch  PreReleas BuildLabel
                     eLabel
-----  -----  -----  --------- ----------
1      0      0      beta1
```
That still doesn't mean we can't leave it at `1.0.0.0` in the repo but perhaps we should think about moving away from this strategy.

Reply:
I wish the newer version of PowerShellGet reported the pre-release info.  If it did, I'd say we leave it at `1.0.0` and set the `Prerelease` private data field to `alpha0`.

Reply:
The issue template includes `PreReleaseVersion`, so as long as that's something dev-ish we should be fine. `1.0.0-dev`?

Reply:
Doesn't `-dev` sort after `-alpha` and `-beta` in semver 1?  That's why I suggested `alpha0` for the checked in version of the psd1 file.  Or we could just make that `alpha` since `alpha` sorts before `alpha0`.

Reply:
I guess my thought was that a local version should always take precedence over a packaged version, but I suppose an actual prerelease of 1.0.1 should take priority over the first commit on the way to 1.0.1? Really not worth overthinking...

But since overthinking is fun, the [SemVer 1.0.0 spec](https://semver.org/spec/v1.0.0.html) says:

> A pre-release version number MAY be denoted by appending an arbitrary string immediately following the patch version and a dash. The string MUST be comprised of only alphanumerics plus dash [0-9A-Za-z-]. Pre-release versions satisfy but have a lower precedence than the associated normal version. Precedence SHOULD be determined by lexicographic ASCII sort order. For instance: 1.0.0-alpha1 < 1.0.0-beta1 < 1.0.0-beta2 < 1.0.0-rc1 < 1.0.0.

So, a `PreReleaseVersion` of `-0` seems to have the absolute lowest precedence, but I'm fine with `-alpha`/`-beta` (no number) indicating dev versions and actual prereleases of `-alpha0`, `-alpha1`, `-beta0`, etc.

Reply:
Updated to set it to `alpha`.  According to docs (and peeking at PSReadline beta) the `-` prefix is not required.